projects/plugins/crm/modules/woo-sync/includes/class-woo-sync.php

```diff
@@ -1831,104 +1833,90 @@ private function generate_site_key_string( $site_url ){
 		$pre_state = $this->skip_local_woo_check;
 		$this->skip_local_woo_check = true;
-		$sync_sites = $this->get_active_sync_sites( 'default' );
+
 		$this->skip_local_woo_check = $pre_state;
 
-        // basic validation
-        if ( !in_array( $mode, array( JPCRM_WOO_SYNC_MODE_LOCAL, JPCRM_WOO_SYNC_MODE_API ) ) ){
-        	return false;
-        }
-        if ( isset( $sync_sites[ $site_key ] ) ){
-        	return false;
-        }
+		// basic validation
+		if ( ! in_array( $mode, array( JPCRM_WOO_SYNC_MODE_LOCAL, JPCRM_WOO_SYNC_MODE_API ), true ) ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
+			return false;
+		}
 
-	    
-        // if no site key, attempt to generate one:
-        if ( empty( $site_key ) ){
+		// if no site key, attempt to generate one:
+		if ( empty( $site_key ) ) {
 
-			if ( $mode == JPCRM_WOO_SYNC_MODE_LOCAL ){
-				
+			if ( $mode === JPCRM_WOO_SYNC_MODE_LOCAL ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
 				$site_key = 'local';
+			} elseif ( ! empty( $domain ) ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
 
-				// if local and already have a local, error?
-		        if ( isset( $sync_sites[ $site_key ] ) ){
-		        	return false;
-		        }
+				$site_key = $this->generate_site_key( $domain ); // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
 
 			} else {
 
-				if ( !empty( $domain ) ){
-					
-					$site_key = $this->generate_site_key( $domain );
-
-				} else {
-
-					// external site setup without a domain ¯\_(ツ)_/¯
-					$site_key = $this->generate_site_key( 'no_domain' );
-
-				}
+				// external site setup without a domain ¯\_(ツ)_/¯
+				$site_key = $this->generate_site_key( 'no_domain' );
 
 			}
 
-			// any luck?
-			if ( empty( $site_key ) ){
+			$sync_sites = $this->get_active_sync_sites( 'default' );
 
+			if (
+				// error generating key
+				empty( $site_key )
+				// site key already exists
+				|| isset( $sync_sites[ $site_key ] )
+				// domain already exists
+				|| in_array( $domain, array_column( $sync_sites, 'domain' ), true ) // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
+				) {
 				return false;
-
 			}
 
 		}
 
-        
-        // add
-        $sync_sites[ $site_key ] = array(
-
-            'mode'           => $mode,
-            'domain'         => $domain,
-            'key'            => $key,
-            'secret'         => $secret,
-            'prefix'         => $prefix,
-
-            // tracking
-            'last_sync_fired'        => -1,
-            'resume_from_page'       => 1,
-            'total_order_count'      => 0,
-            'total_customer_count'   => 0,
-            'first_import_complete'  => false,
+		// add
+		$sync_sites[ $site_key ] = array(
+			'mode'                  => $mode, // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
```

Comment:
Small thing, but would it not look cleaner if we removed the duplication here then above say line 1857 add // phpcs:disable VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable

Adding it back ( // phpcs:enable) after line 1911? There are 12 instances within 50ish lines otherwise.

Reply:
Done in 02e3386d91.

Even though they're less "clean", typically I prefer per-line ignores, as disabling a rule for a whole section would prevent detection of new code violations within said section. In this case, however, the specific rule violation is so prevalent I'm okay with disabling it for the section (plus I was forced to disable PHPCS for the whole file anyway).